rspace/src/main/scala/coop/rchain/rspace/pure/StoreInstances.scala

```diff
@@ -0,0 +1,411 @@
+package coop.rchain.rspace.pure
+
+import java.nio.ByteBuffer
+
+import cats.data.{ReaderT}
+import coop.rchain.catscontrib.Capture
+import coop.rchain.rspace.Serialize
+import coop.rchain.rspace.internal._
+import org.lmdbjava.{CursorIterator, KeyRange, Txn}
+import java.security.MessageDigest
+
+import cats.Monad
+import coop.rchain.rspace.util.{ignore, withResource}
+import coop.rchain.rspace.internal.scodecs._
+import scodec.Codec
+import scodec.bits.{BitVector, ByteVector}
+
+import scala.collection.immutable.Seq
+import coop.rchain.rspace.util
+
+import scala.collection.JavaConverters._
+
+object StoreInstances {
+
+  implicit def storeLMDB[F[_]: Monad, C, P, A, K](
```

Comment:
As mentioned before, I'd prefer to do the explicit desugaring and add

```scala
monadF: Monad[F],
```

to the implicit parameter list.